jobs/integration/utils.py

```diff
@@ -199,12 +200,13 @@ async def retry_async_with_timeout(
 
     """
     deadline = time.time() + timeout_insec
+    results = None
```

Comment:
the async `func` can return anything that is truthy.  objects can be truthy if they implement the `__bool__` magic method.  Knowing this, the `timeout_msg` could have a format marker to print the timed out object. 

consider this:

```python

class Truthy:
    def __bool__(self):
        return False

    def __repr__(self):
        return "Truthy()"

async def my_async():
    return Truthy()


...
    await retry_async_with_timeout(
        my_async,
        tuple(),
        timeout_msg="I will timeout because my_async returns {}"
    )
```


This will eventually fail with the expected TimeoutError
but that Exception will have some test describing the timeout exception
`I will timeout because my_async returns Truthy()`